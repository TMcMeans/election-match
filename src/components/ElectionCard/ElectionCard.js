import React from 'react';
import './ElectionCard.css';

export const ElectionCard = props => {
  let date = new Date(props.date).toString().substring(0, 15);

  return (
    <div className="election-card">
      <h3>{props.description}</h3>
      <h3>{date}</h3>
      <a href={props['polling-place-url']}>Your polling website</a>
    </div>
  );
};

//Election object descriptors:
// the jurisdiction it affects (defined using an Open Civic Data Identifier, or OCD-ID)
// a type label (e.g. - municipal, special, school)
// the deadline to register to vote
// the deadline to request an absentee ballot
// whether or not same-day voter registration is available
// whether or not absentee voting requires an excuse
// (for primary elections) whether participation is open to voters without a party affiliation, or closed to party members only
