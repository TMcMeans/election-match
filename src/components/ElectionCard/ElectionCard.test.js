import React from 'react';
import { shallow } from 'enzyme';

import { ElectionCard } from './ElectionCard';

describe('ElectionCard', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<ElectionCard />);
    expect(wrapper).toMatchSnapshot();
  });
});
