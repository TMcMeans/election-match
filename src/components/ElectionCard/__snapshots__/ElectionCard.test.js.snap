// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ElectionCard should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ElectionCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3 />,
        <h3>
          Invalid Date
        </h3>,
        <a>
          Your polling website
        </a>,
      ],
      "className": "election-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Invalid Date",
        },
        "ref": null,
        "rendered": "Invalid Date",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Your polling website",
          "href": undefined,
        },
        "ref": null,
        "rendered": "Your polling website",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3 />,
          <h3>
            Invalid Date
          </h3>,
          <a>
            Your polling website
          </a>,
        ],
        "className": "election-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Invalid Date",
          },
          "ref": null,
          "rendered": "Invalid Date",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Your polling website",
            "href": undefined,
          },
          "ref": null,
          "rendered": "Your polling website",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
